### PR TITLE
[GEOT-7361] Support non EPSG authorities in DataUtilities type spec, and property data store

### PR DESCRIPTION
**Pull request by aaime**
[GEOT-7361](https://osgeo-org.atlassian.net/browse/GEOT-7361) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

Allow property data store to use the IAU authority, and other authorities, in support for non EPSG authority code development and test (as the property store is our favourite test data store).

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: modules/library/main/src/main/java/org/geotools/data/DataUtilities.java [hunk 1]
```diff
@@ -1844,17 +1844,21 @@ private static void setDefaultGeometry(
      *
      *     <ul>
      *       <li><code>nillable</code>
+     *       <li><code>authority=<#></code>
```
Comment (jodygarnett) [inline modules/library/main/src/main/java/org/geotools/data/DataUtilities.java:1847]:
```suggestion
     *       <li><code>authority=text</code>
```

File: modules/library/main/src/main/java/org/geotools/data/DataUtilities.java [hunk 2]
```diff
@@ -1844,17 +1844,21 @@ private static void setDefaultGeometry(
...
      *       <li><code>srid=<#></code>
```
Comment (jodygarnett) [inline modules/library/main/src/main/java/org/geotools/data/DataUtilities.java:1848]:
```suggestion
     *       <li><code>srid=number</code>
```

File: modules/library/main/src/main/java/org/geotools/data/DataUtilities.java [hunk 3]
```diff
@@ -1844,17 +1844,21 @@ private static void setDefaultGeometry(
...
      *     </ul>
      *
      * <li>You may indicate the default Geometry with an astrix: "*geom:Geometry".
-     * <li>You may also indicate the srid (used to look up a EPSG code): "*point:Point:3226
+     * <li>You may also indicate the srid (used to look up a EPSG code): "*point:Point:3226"
+     * <li>The CRS authority can be specified using the <code>authority</code> hint, if it's not
+     *     EPSG (the default): "*point:Point:authority=IAUsrid=49900"
```
Comment (jodygarnett) [inline modules/library/main/src/main/java/org/geotools/data/DataUtilities.java:1854]:
```suggestion
     *     EPSG (the default): "*point:Point:authority=IAU:srid=49900"
```

**Review by jodygarnett — CHANGES_REQUESTED** [review]
Some minor doc changes

**Comment by aaime** [comment]
All suggested changes applied, thanks @jodygarnett 
Merging.

**Pull request merged**